### PR TITLE
librustdoc: blur page when help dialogue is present

### PR DESCRIPTION
**Pull request by andreastt**
Blurs the document's background when the help dialogue is present,
making the contents of the dialogue stand out more.

### CODE REVIEW

**Comment by steveklabnik** [comment]
I like this idea, and think it looks fine, but am terrible at CSS and would like @Gankro or @alexcrichton to check it too.

**Comment by andreastt** [comment]
I’m sure it can be done in cleverer ways.  I just realised this patch also fixes the positioning of the dialogue as you scroll.  It relies on flexbox to centre the box vertically as well as horizontally.

**Comment by alexcrichton** [comment]
Could you post this on a site to poke around on? Also is this supported in most browsers?

File: src/librustdoc/html/static/main.css [hunk 1]
```diff
@@ -445,20 +445,27 @@ a {
 
 tr.result span.primitive::after { content: ' (primitive type)'; font-style: italic; }
 
+body.blur > :not(#help) { filter: blur(8px); opacity: .7; }
+
 #help {
+    width: 100%;
+    height: 100vh;
```
Comment (Gankro) [inline src/librustdoc/html/static/main.css]: mind: blown

**Comment by nikomatsakis** [comment]
Everything I know about HTML and CSS I learned in 1997. r? @alexcrichton

**Comment by andreastt** [comment]
@alexcrichton 

Before:
Before
After:
After

**Comment by alexcrichton** [comment]
@bors: r+ 0baadfc34e6afc679043d280a1c6b7b47bf3c8a4

Nice!

**Comment by P1start** [comment]
According to [caniuse](http://caniuse.com/#feat=css-filters), `filter` requires a `-webkit-` prefix to support Chrome (and other WebKit browsers).

**Comment by steveklabnik** [comment]
@andreastt needs a rebase :)

**Comment by andreastt** [comment]
@steveklabnik Just got back from holiday 🌴

**Comment by andreastt** [comment]
@P1start Thanks, added the prefix!

**Comment by steveklabnik** [comment]
@andreastt nice, hope you had a good time :)

**Pull request merged**